### PR TITLE
[INLONG-4273][Manager] Support MongoDB source

### LINKED ISSUE #4273: [Feature][Manager] Add MongoDB source support
### Description

Add MongoDB source support, and add extract node utils.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by LvJiancheng**
### Title Name: [INLONG-4273] [Manager] support Mongo source

where *XYZ* should be replaced by the actual issue number.

Fixes #4273 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Review by EMsnap — APPROVED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/mongo/MongoSourceOperation.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ */
+@Service
+public class MongoSourceOperation extends AbstractSourceOperation {
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public Boolean accept(SourceType sourceType) {
+        return SourceType.KAFKA == sourceType;
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/mongo/MongoSourceOperation.java:54]: Change all of `Kafka` to `Mongo`, please.

**Review by healchow — CHANGES_REQUESTED** [review]

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/mongo/MongoSourceOperation.java (hunk 1 above)
Comment (LvJiancheng) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/source/mongo/MongoSourceOperation.java]: done

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java [hunk 2]
```diff
@@ -19,20 +19,28 @@
...
 import org.apache.inlong.manager.common.enums.SourceType;
-import org.apache.inlong.manager.common.pojo.source.SourceResponse;
+import org.apache.inlong.manager.common.pojo.source.SourceRequest;
+import org.apache.inlong.manager.common.pojo.source.StreamSource;
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.common.util.JsonTypeDefine;
 
 /**
  * Response of the kafka source
  */
```
Comment (gong) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java:35]: describe error

Comment (LvJiancheng) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java]: done

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSourceDTO.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                .primaryKey(request.getPrimaryKey())
+                .hosts(request.getHosts())
+                .username(request.getUsername())
+                .password(request.getPassword())
+                .database(request.getDatabase())
+                .collection(request.getCollection())
+                .build();
+    }
+
+    public static MongoSourceDTO getFromJson(@NotNull String extParams) {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSourceDTO.java:75]: Add java doc please.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+ *  Mongo source info
+ */
+@Data
+@SuperBuilder
+@AllArgsConstructor
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+@ApiModel(value = "Mongo source info")
+@JsonTypeDefine(value = SourceType.SOURCE_MONGO)
+public class MongoSource extends StreamSource {
```
Comment (healchow) [inline inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java:43]: It is recommended to call the `MongoDB`, which is the official name.

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSourceDTO.java (hunk 3 above)
Comment (LvJiancheng) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSourceDTO.java]: done

File: inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java (hunk 4 above)
Comment (LvJiancheng) [reply inlong-manager/manager-common/src/main/java/org/apache/inlong/manager/common/pojo/source/mongo/MongoSource.java]: ok ,good idea

**Pull request merged**